src/AmqpBundle/Amqp/DataCollector.php

```diff
@@ -72,33 +72,36 @@ public function getCommands()
      *
      * @return string data collector name
      */
-    public function getName()
+    public function getName(): string
     {
         return $this->name;
     }
 
     /**
-     * Temps total d'exec des commandes.
+     * Return total command execution time.
      *
      * @return float
      */
-    public function getTotalExecutionTime()
+    public function getTotalExecutionTime(): float
     {
-        $ret = 0;
-        foreach ($this->data['commands'] as $command) {
-            $ret += $command['executiontime'];
-        }
-
-        return $ret;
+        return (float) array_sum(array_column($this->data['commands'], 'executiontime'));
     }
 
     /**
-     * Temps moyen d'exec.
+     * Get average execution time.
      *
      * @return float
      */
-    public function getAvgExecutionTime()
+    public function getAvgExecutionTime(): float
+    {
+        return $this->getTotalExecutionTime() ? ($this->getTotalExecutionTime() / \count($this->data['commands'])) : (float) 0;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reset()
     {
-        return ($this->getTotalExecutionTime()) ? ($this->getTotalExecutionTime() / count($this->data['commands'])) : 0;
+        $this->data = [['commands' => []];
```

Comment:
Don't you need to call `parent::reset()` to ensure proper `$this-data` reset? 🤔 
```php
parent::reset();
$this->data['commands'] = [];
```
I didn't find where is defined `data` and who is in charge of its default initialization 😕

Reply:
Not really, most other data collectors I ran into so far do just reset to empty array, as do most of the official ones from symfony itself.

Reply:
Sounds good 😃 
Let's go for `v3.0.1` 🎉